src/legacy/core_plugins/console/public/src/components/settings_modal.tsx

```diff
@@ -0,0 +1,263 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Fragment, useState, useEffect } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldNumber,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+} from '@elastic/eui';
+import { DevToolsSettings } from './dev_tools_settings';
+
+interface Props {
+  onSaveSettings: (newSettings: DevToolsSettings) => Promise<void>;
+  onClose: () => void;
+  refreshAutocompleteSettings: () => void;
+  settings: DevToolsSettings;
+}
+
+export function DevToolsSettingsModal(props: Props) {
+  const [fontSize, setFontSize] = useState(props.settings.fontSize);
+  const [wrapMode, setWrapMode] = useState(props.settings.wrapMode);
+  const [fields, setFields] = useState(props.settings.autocomplete.fields);
+  const [indices, setIndices] = useState(props.settings.autocomplete.indices);
+  const [templates, setTemplates] = useState(props.settings.autocomplete.templates);
+  const [polling, setPolling] = useState(props.settings.polling);
+  const [tripleQuotes, setTripleQuotes] = useState(props.settings.tripleQuotes);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    return () => {
+      // TODO
+    };
+  }, []);
+
+  function saveSettings() {
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+    props.onSaveSettings({
+      fontSize,
+      wrapMode,
+      autocomplete: {
+        fields,
+        indices,
+        templates,
+      },
+      polling,
+      tripleQuotes,
+    });
+
+    setIsLoading(false);
+  }
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal
+        data-test-subj="devToolsSettingsModal"
+        className="devToolsSettingsModal"
+        onClose={props.onClose}
+      >
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage id="console.settingsPage.pageTitle" defaultMessage="Settings" />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody className="euiText euiText--medium euiText--constrainedWidth">
+          <EuiFlexGrid columns={1}>
+            <EuiFlexItem>
+              <EuiFlexItem>
+                <EuiText>
+                  <p>
+                    <FormattedMessage
+                      id="console.settingsPage.fontSizeLabel"
+                      defaultMessage="Font Size"
+                    />
+                  </p>
+                </EuiText>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiFieldNumber
```

Comment:
This should be wrapped in a `<EuiFormRow label="Font size">`.  See http://localhost:8030/#/forms/form-layouts for more info on that component.